src/core/variable.jl

```diff
@@ -469,6 +446,66 @@ function variable_branch_current_rectangular(pm::AbstractPowerModel; nw::Int=pm.
     branch = ref(pm, nw, :branch)
     bus = ref(pm, nw, :bus)
 
+    if bounded
+        ub = Dict()
+            for (l,i,j) in ref(pm, nw, :arcs_from)
+            b = branch[l]
+            ratefr = b["rate_a"][cnd]*b["tap"][cnd]
+            rateto = b["rate_a"][cnd]
+            ub[l]  = max(ratefr/bus[i]["vmin"][cnd], rateto/bus[j]["vmin"][cnd])
+        end
+        var(pm, nw, cnd)[:cr] = JuMP.@variable(pm.model,
+            [(l,i,j) in ref(pm, nw, :arcs)], base_name="$(nw)_$(cnd)_cr",
+            lower_bound = -ub[l],
+            upper_bound = ub[l],
+            start = comp_start_value(ref(pm, nw, :branch, l), "cr_start", cnd)
+        )
+        var(pm, nw, cnd)[:ci] = JuMP.@variable(pm.model,
+            [(l,i,j) in ref(pm, nw, :arcs)], base_name="$(nw)_$(cnd)_ci",
+            lower_bound = -ub[l],
+            upper_bound = ub[l],
+            start = comp_start_value(ref(pm, nw, :branch, l), "ci_start", cnd)
+        )
+
+    else
+        var(pm, nw, cnd)[:cr] = JuMP.@variable(pm.model,
+            [(l,i,j) in ref(pm, nw, :arcs)], base_name="$(nw)_$(cnd)_cr",
+            start = comp_start_value(ref(pm, nw, :branch, l), "cr_start", cnd)
+        )
+        var(pm, nw, cnd)[:ci] = JuMP.@variable(pm.model,
+            [(l,i,j) in ref(pm, nw, :arcs)], base_name="$(nw)_$(cnd)_ci",
+            start = comp_start_value(ref(pm, nw, :branch, l), "ci_start", cnd)
+        )
+    end
+
+    # store expressions in rectangular power variable space
+    p = Dict()
+    q = Dict()
+
+    for (l,i,j) in ref(pm, nw, :arcs_from)
+        vrf = var(pm, nw, cnd, :vr, i)
+        vif = var(pm, nw, cnd, :vi, i)
+        crf = var(pm, nw, cnd, :cr, (l,i,j))
+        cif = var(pm, nw, cnd, :ci, (l,i,j))
+
+        vrt = var(pm, nw, cnd, :vr, j)
+        vit = var(pm, nw, cnd, :vi, j)
+        crt = var(pm, nw, cnd, :cr, (l,j,i))
+        cit = var(pm, nw, cnd, :ci, (l,j,i))
+        p[(l,i,j)] = vrf*crf  + vif*cif
+        q[(l,i,j)] = vif*crf  - vrf*cif
+        p[(l,j,i)] = vrt*crt  + vit*cit
+        q[(l,j,i)] = vit*crt  - vrt*cit
+    end
+    var(pm, nw, cnd)[:p] = p
+    var(pm, nw, cnd)[:q] = q
+end
+
+
+"variable: `csr[l,i,j] + j csi[l,i,j] ` for `(l,i,j)` in `arcs_from`"
+function variable_branch_series_current_rectangular(pm::AbstractPowerModel; nw::Int=pm.cnw, cnd::Int=pm.ccnd, bounded = true)

```

Comment:
It seems odd to me to include the `_rectangular` qualifier here.  Shouldn't the model type be encoding this?

Reply:
The functions in variable.jl do typically not dispatch based on the actual powermodel type, for re-usability, e.g.
```
function variable_branch_current_rectangular(pm::AbstractPowerModel; nw::Int=pm.cnw, cnd::Int=pm.ccnd, bounded = true)
function variable_branch_series_current_rectangular(pm::AbstractPowerModel; nw::Int=pm.cnw, cnd::Int=pm.ccnd, bounded = true)
```
I can split them up into a real and imaginary part if you prefer that?

 In the current implementation, you have a from and to side total current variable, and a single (from-side) series current variable. Therefore, in iv.jl there is :
```
function variable_branch_current(pm::AbstractIVRModel; kwargs...)
    variable_branch_current_rectangular(pm; kwargs...)
    variable_branch_series_current_rectangular(pm; kwargs...)
end
```
 I can change that to : 
 ```
function variable_branch_current(pm::AbstractIVRModel; kwargs...)
    variable_branch_total_current(pm; kwargs...)
    variable_branch_series_current(pm; kwargs...)
end
```
but that would lead to potential confusion if you ever add `variable_branch_current_polar`?

Reply:
Following the pattern from voltages in ACP and ACR.  A meta variable definition (e.g. `variable_voltage`) dispatches to different formulation agnostic variable definitions, e.g. `variable_voltage_angle `, `variable_voltage_magnitude ` / `variable_voltage_real`, `variable_voltage_imaginary`.  It is possible to follow a similar pattern here?

Reply:
Ok, I separated real and imaginary and put the construction of the active/reactive power expressions under `variable_branch_current`